include/LuaContext.hpp

```diff
@@ -1615,11 +1615,10 @@ class LuaContext {
             throw WrongTypeException(lua_typename(state, index), typeid(TFirstType));
 
         Binder<TCallback, const TFirstType&> binder{ callback, *firstElem };
-        return readIntoFunction(state, retValueTag, binder, index + 1, othersTags...);
+        return readIntoFunction(state, retValueTag, binder, index + 1, tag<TTypes...>());
     }
-    template<typename TRetValue, typename TCallback, typename TFirstType, typename... TTypes>
-    static auto readIntoFunction(lua_State* state, tag<TRetValue> retValueTag, TCallback&& callback, int index, tag<TFirstType>, tag<TTypes>... othersTags)
-        -> typename std::enable_if<!IsOptional<TFirstType>::value, TRetValue>::type
```

Comment:
Was there a reason why you moved the `enable_if`? `, TRetValue` should be unnecessary in this case, the `void` default is fine there.